app/routes/surveys/AddSurveyRoute.js

```diff
@@ -1,31 +1,85 @@
 import { connect } from 'react-redux';
 import { compose } from 'redux';
-import { addSurvey, deleteSurvey } from '../../actions/SurveyActions';
+import {
+  addSurvey,
+  deleteSurvey,
+  fetchTemplate
+} from '../../actions/SurveyActions';
 import SurveyEditor from './components/SurveyEditor/SurveyEditor';
 import { LoginPage } from 'app/components/LoginForm';
 import replaceUnlessLoggedIn from 'app/utils/replaceUnlessLoggedIn';
 import { push } from 'react-router-redux';
-import moment from 'moment-timezone';
 import { initialQuestion } from './components/SurveyEditor/SurveyEditor';
+import prepare from 'app/utils/prepare';
+import { selectSurveyTemplate } from 'app/reducers/surveys';
+import { fetchEvent } from 'app/actions/EventActions';
+import { selectEventById } from 'app/reducers/events';
+import { defaultActiveFrom } from './utils';
 
-const time = (hours, minutes) =>
-  moment()
-    .startOf('day')
-    .add({ hours, minutes })
-    .toISOString();
+const loadData = (props, dispatch) => {
+  const { template, event } = props.location.query;
+  if (event) {
+    return dispatch(fetchEvent(event)).then(
+      result =>
+        result.success &&
+        dispatch(fetchTemplate(result.payload.entities.events[event].eventType))
+    );
+  }
+  if (template) {
+    return dispatch(fetchTemplate(template));
+  }
+};
 
 const mapStateToProps = (state, props) => {
+  const notFetching = !state.surveys.fetching && !state.events.fetching;
+  const { templateType, event } = props.location.query;
+
+  const fullEvent = selectEventById(state, { eventId: event });
+
+  const template = selectSurveyTemplate(state, {
+    ...props,
+    templateType: templateType || fullEvent.eventType
+  });
+
+  const initialEvent = event
+    ? {
+        value: fullEvent.id,
+        label: fullEvent.title
+      }
+    : '';
+  const activeFrom = event ? fullEvent.endTime : defaultActiveFrom(12, 0);
+
+  let initialValues = null;
+  if (
+    !(
+      ((templateType || event) && !template) ||
+      (event && Object.keys(fullEvent).length === 0)
+    )
+  ) {
+    if (template) {
+      initialValues = {
+        ...template,
+        event: initialEvent,
+        activeFrom
+      };
+    } else {
+      initialValues = {
+        activeFrom,
+        event: initialEvent,
+        title: '',
+        questions: [initialQuestion]
+      };
+    }
+  }
+
   return {
-    initialValues: {
-      activeFrom: time(12, 0),
-      event: '',
-      title: '',
-      isClone: true,
-      questions: [initialQuestion]
-    },
+    template,
+    initialValues,
     survey: {
-      questions: []
-    }
+      questions: template ? template.questions : [],
+      event: event && fullEvent
```

Comment:
couldn't this just be `event: fullEvent`?

Reply:
The event selector returns an empty object if it doesn't find anything, and I think I do some `if (event) { ... }` later on which would evaluate to true. I didn't want to deal with any `object 'event' has no attribute 'title'` error messages. Think that's bad practise?